File: Data Structures & Algorithms/jump-game-ii/submission-3.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        res = 0
        left, right = 0,0
        while right < len(nums)-1:
            far = 0
            for i in range(left, right +1):
                far = max(far, i+ nums[i])
            left = right + 1
            right = far # this would be a jump
            res += 1
        return res
        

File: Data Structures & Algorithms/maximum-subarray/submission-3.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int prefix = 0;
        int maxS = nums[0];
        for(int i = 0; i < nums.size(); i++){
            if(prefix < 0) prefix = 0;
            prefix += nums[i];
            maxS = max(maxS,prefix);
        }
        return maxS;
    }
};
